### PR TITLE
Add editorial checks through Vale

### LINKED ISSUE #8: Add editorial style linting checks
### Existing issue check

- [X] I found no existing [issues](https://github.com/software-gardening/almanac/issues) covering this topic.

### Description

Writing often benefits from checks on grammar and other common challenges (such as repeated words, paragraphs which are too long, linguistic complexity, etc). This issue highlights the need to add a tool which can check for this as time goes on. There may be others, but I've found https://vale.sh to be a good group of checks with reasonable support.

### Example

n/a

### Would you like to work on a solution for this?

- [X] Yes I am willing to submit a PR!

### PR DESCRIPTION
**Pull request by d33bs**
# Description

This PR adds a number of new editorial checks through [Vale](https://vale.sh/). It also wraps the existing checks through Alex.js. Vale is installed as a Python-based Poetry dev group dependency and used through pre-commit for this project.

Note: I found that in order to have this run as a pre-commit check with warnings included I needed a short script, which has been added as a Poe the Poet task. It seems this goes against the designs of Vale's intention (see https://github.com/errata-ai/vale/issues/575). Just the same, it felt to me important that warnings be enough to trigger a pre-commit failure.

Closes #8 

## What is the nature of your change?

- [ ] Content additions or updates (adds or updates content)
- [ ] Bug fix (fixes an issue).
- [x] Enhancement (adds functionality).
- [ ] Breaking change (these changes would cause existing functionality to not work as expected).

# Checklist

Please ensure that all boxes are checked before indicating that this pull request is ready for review.

- [x] I have read the [CONTRIBUTING.md](CONTRIBUTING.md) guidelines.
- [x] My code follows the style guidelines of this project.
- [x] I have performed a self-review of my own contributions.
- [x] I have commented my content, particularly in hard-to-understand areas.
- [x] I have made corresponding changes to related documentation (outside of book content).
- [x] My changes generate no new warnings.
- [x] New and existing tests pass locally with my changes.
- [ ] I have added tests that prove my additions are effective or that my feature works.
- [x] I have deleted all non-relevant text in this pull request template.

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -33,6 +34,15 @@ build-book-dev.shell = """
...
 """
+# runs vale relative to this project through pre-commit
+# to trigger warnings as non-zero return when there is
+# more than one line of output from vale.
+vale-checks.shell = """
+output=$(poetry run vale ./src/almanac) && \
+if [ $(echo $output | wc -l) -gt 1 ]; then \
+echo $output && exit 1; \
+fi
+"""
```
Comment (falquaddoomi) [inline pyproject.toml:45]: I'd noticed when I tried to run `pre-commit run --all-files` that this command  was failing with the message `[loadStyles] Runtime error style 'proselint' does not exist on StylesPath`. I took me a while to figure out, by reading the GH workflow, that I should have run `poetry run vale sync` to pull the vale styles(?). Perhaps adding it here, or at least instructing the user to run it, would be helpful?

**Review by falquaddoomi — COMMENTED** [review]
I had a minor issue with running the pre-commit check, but otherwise this looks good!

File: pyproject.toml (hunk 1 above)
Comment (d33bs) [reply pyproject.toml:45]: Thanks for this, great point! I'll add another part to this shell command to sync before running Vale.

**Comment by d33bs** [comment]
Thanks @falquaddoomi ! I've made a change to enable vale to sync before checking work through pre-commit, and will now merge.

**Review by falquaddoomi — APPROVED** [review]

**Comment by d33bs** [comment]
Thanks again @falquaddoomi !

**Pull request merged**